### ISSUE TITLE
💅 noExtraBooleanCast fix should not remove braces

### ISSUE THREAD

**Issue by Sam-Apostel** [biomejs/biome#7225]
### Environment information

```bash
(happens in playground on default settings
```

### Rule name

lint/complexity/noExtraBooleanCast

### Playground link

https://biomejs.dev/playground/?code=YwBvAG4AcwB0ACAAYgAwACAAPQAgAGYAYQBsAHMAZQA7AAoAYwBvAG4AcwB0ACAAYgAxACAAPQAgAGYAYQBsAHMAZQAKAAoAYwBvAG4AcwB0ACAAYgBvAG8AbABlAGEAbgAgAD0AIAAhAEIAbwBvAGwAZQBhAG4AKABiADAAIAAmACYAIABiADEAKQA7AAoACgAvAC8AIABiAGUAZgBvAHIAZQAgAGYAaQB4ADoAIABiAG8AbwBsAGUAYQBuACAAaQBzACAAdAByAHUAZQAKAC8ALwAgAGEAZgB0AGUAcgAgAGYAaQB4ADoAIABiAG8AbwBsAGUAYQBuACAAaQBzACAAZgBhAGwAcwBlAA%3D%3D&language=ts

### Expected result

In the playground, look at the `Analyzer Fixes` tab to see the fixed output.

## Expected Result
`!Boolean(b0 && b1)` is fixed as `!(b0 && b1)`

## Actual result
`!Boolean(b0 && b1)` is fixed as `!b0 && b1`

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by JeetuSuthar** [comment]
I'm working on this issue
